src/cmd/cli/command/mcpClient.go

```diff
@@ -0,0 +1,128 @@
+package command
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/DefangLabs/defang/src/pkg/term"
+)
+
+// ClientProcessMap maps client names to their process names
+var ClientProcessMap = map[string]string{
+	"claude":   "Claude",
+	"windsurf": "Codeium",
+	"cursor":   "Cursor",
+	"vscode":   "Code",
+}
+
+// IsClientRunning checks if a client process is currently running
+func IsClientRunning(client string) bool {
+	processName := getProcessName(client)
+
+	switch runtime.GOOS {
+	case "windows":
+		cmd := exec.Command("tasklist", "/FI", "IMAGENAME eq "+processName+".exe", "/NH")
+		output, err := cmd.Output()
+		if err != nil {
+			return false
+		}
+		return strings.Contains(string(output), processName+".exe")
+	case "darwin":
+		cmd := exec.Command("pgrep", "-x", processName)
+		output, err := cmd.Output()
+		if err != nil {
+			return false
+		}
+		return strings.TrimSpace(string(output)) != ""
+	case "linux":
+		cmd := exec.Command("pgrep", "-f", strings.ToLower(processName))
+		output, err := cmd.Output()
+		if err != nil {
+			return false
+		}
+		return strings.TrimSpace(string(output)) != ""
+	default:
+		return false
+	}
+}
+
+// RestartClient attempts to restart the specified client application
+func RestartClient(client string) error {
```

Comment:
This feels risky.  Aren't we potentially destroying unpersisted application state?

Reply:
done!